packages/legend-application-query/src/stores/QueryBuilderProjectionState.ts

```diff
@@ -795,17 +795,26 @@ export class QueryBuilderProjectionState {
   }
 
   isValidProjectionState(): boolean {
```

Comment:
I don't think this is a good idea, since you would then have to do the awkward check of `!queryBuilderState.isValidQueryBuilderState() && !queryBuilderState.isDuplicatedQueryBuilderState()`, instead, we can create a separate method to return a validation, checkout `ValidationHelper`. The method `isValidProjectionState` can then simply be a method that checks if we have any validation issues or not.